### ISSUE TITLE
about codebook

### ISSUE THREAD

**Issue by Dorniwang** [openai/DALL-E#52]
so in the whole architecture, no explicit codebook vecters are used right? only categorical logits as the input to the decoder when you train your dvae?

**Comment by Asthestarsfalll** [comment]
same question.

**Comment by andras-simonyi** [comment]
Late to the party, but had the same question... The answer, I think, is that the decoder starts with a 1x1 convolution, which is equivalent to having a learned codebook, and using the latent probability-weighted sum of the codebook vectors as a starting point for decoding.
